src/node_report_module.cc

```diff
@@ -0,0 +1,539 @@
+
+  // If report newly requested for fatalerror, set up the V8 callback
+  if ((nodereport_events & kReportOnFatalError) &&
+      (id->error_hook_initialised == false)) {
+    isolate->SetFatalErrorHandler(OnFatalError);
+    id->error_hook_initialised = true;
+  }
+
+  if (nodereport_events & kReportOnException) {
+    id->exception_hook_initialised = true;
+  }
+
+#ifndef _WIN32
+  // If report newly requested on external user signal
+  // set up watchdog thread and handler
+  if ((nodereport_events & kReportOnSignal) &&
+      (id->signal_thread_initialised == false)) {
+    SetupSignalHandler();
+  }
+  // If report no longer required on external user signal,
+  // reset the OS signal handler
+  if (!(nodereport_events & kReportOnSignal) &&
+       (previous_events & kReportOnSignal)) {
+    RestoreSignalHandler(nodereport_signal, &saved_sa);
+  }
+#endif
+}
+
+void SetSignal(const FunctionCallbackInfo<Value>& info) {
+#ifndef _WIN32
+  CHECK(info[0]->IsString());
+  String::Utf8Value parameter(info.GetIsolate(), info[0]);
+  CHECK_NOT_NULL(*parameter);
+  unsigned int previous_signal = nodereport_signal;  // save previous setting
+  nodereport_signal = ProcessNodeReportSignal(*parameter);
+
+  // If signal event active and selected signal has changed,
+  // switch the OS signal handler
+  if ((nodereport_events & kReportOnSignal) &&
+      (nodereport_signal != previous_signal)) {
+    RestoreSignalHandler(previous_signal, &saved_sa);
+    node::RegisterSignalHandler(nodereport_signal, SignalDump, true);
+  }
+#endif
+}
+
+void SetFileName(const FunctionCallbackInfo<Value>& info) {
+  Isolate* isolate = info.GetIsolate();
+  Environment* env = Environment::GetCurrent(isolate);
+  CHECK_NE(env, nullptr);
+  CHECK(info[0]->IsString());
+  String::Utf8Value parameter(isolate, info[0]);
+  CHECK_NOT_NULL(*parameter);
+  ProcessNodeReportFileName(env, *parameter);
+}
+
+void SetDirectory(const FunctionCallbackInfo<Value>& info) {
+  Isolate* isolate = info.GetIsolate();
+  Environment* env = Environment::GetCurrent(isolate);
+  CHECK_NE(env, nullptr);
+  CHECK(info[0]->IsString());
+  String::Utf8Value parameter(isolate, info[0]);
+  CHECK_NOT_NULL(*parameter);
+  ProcessNodeReportDirectory(env, *parameter);
+}
+
+void SetVerbose(const FunctionCallbackInfo<Value>& info) {
+  CHECK(info[0]->IsString());
+  String::Utf8Value parameter(info.GetIsolate(), info[0]);
+  CHECK_NOT_NULL(*parameter);
+  nodereport_verbose = ProcessNodeReportVerboseSwitch(*parameter);
+}
+
+/*******************************************************************************
+ * Callbacks for triggering report on fatal error, uncaught exception and
+ * external signals
+ ******************************************************************************/
+static void OnFatalError(const char* location, const char* message) {
+  Isolate* isolate = Isolate::GetCurrent();
+  Environment* env = Environment::GetCurrent(isolate);
+  CHECK_NE(env, nullptr);
```

Comment:
yes! I just came across this situation while running the tests - upon out of memory conditions when this is invoked, the `Environment::GetCurrent(isolate);` does not exist and crashes program.

So I guess I drop `CHECK_NE`, and allow such scenarios within `TriggerNodeReport` by creating a report through making use of default input.

But how do I check if `env` is NULL? when the invocation itself fails? From a JS binding context I guess we could use use `Environment::GetCurrent(FunctionCallbackInfo)` but here how do we obtain one? probably don't obtain at all, and assume NULL always?

Reply:
@gireeshpunathil Okay, so… we had a longer discussion around these scenarios before in https://github.com/nodejs/node/pull/19134. It looks like, at this point, it’s more complicated than checking the return value of `Environment::GetCurrent(isolate)` for `nullptr`, though. I’ll open a PR today or tomorrow that would introduce that behavior (because I think it makes sense) and ping you in it.

Reply:
Okay, PR up @ https://github.com/nodejs/node/pull/22819

Reply:
thanks!

Reply:
added a TODO at this location that needs to be addressed post - 
https://github.com/nodejs/node/pull/22819 landing

Reply:
the TODO was added but removed when #22819 was landed and leveraged in this area of code